text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+```
+
+There are per replica configurations that are available. If any of these are undefined in the replica and
+defined in the table level props, then those are copied over to the replica configuration.
+But, if the replica has any configuration that is defined and the tabel level properties are defined as well,
+then the replica property value takes precedence and would be assigned.
+
+Options:
+
+* `region` --> Needs to be specified
+* `contributorInsightsEnabled` --> Gets copied over from table level props if defined.
+* `deletionProtection` --> Gets copied over from table level props if defined.
+* `pointInTimeRecovery` --> Gets copied over from table level props if defined.
+* `tableClass` --> Gets copied over from table level props if defined.
+* `tags` --> Gets copied over from table level props if defined.
+* `readWithAutoScaling` --> Gets copied over from table level props if defined.
+* `readCapactity` --> Gets copied over from table level props if defined.
+* `encryptionKey` --> This needs to be defined per replica
+* `kinesisStream` --> This needs to be defined per replica
+* `gsiProps`
+  * `indexName` --> Needs to be specified
+  * `contributorInsightsEnabled` --> Gets copied over from table level props or replica level props if defined.
+
+You can also add a replica using `addReplica` method.
+
+```typescript
+  globalTable.addReplica({
+    region: 'us-east-1',
+  });
+```
+
+__NOTE:__
+
+* A replica needs to be deployed to the region where stack is deployed to. We do this automatically if
+user has not defined the region in replica.
+
+#### Grants
+
+You can grant access to certain operation on the table with the help of `grant` functions.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    replicas: [
+      {
+        region: 'us-west-2',
+      },
+    ],
+  });
+
+  const lambdaRole = new iam.Role(stack, 'Role', {
+    assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+  });
+
+  globalTable.grantReadData(lambdaRole, ['us-west-2']);
+```
+
+#### Metrics
+
+You can access metrics emitted by global table by using `metric` functions.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    replicas: [
+      {
+        region: 'us-west-2',
+      },
+    ],
+  });
+
+  const metric = globalTable.metricConsumedWriteCapacityUnits({
+    region: 'us-west-2',
+  });
+```
+
+#### Importing
+
+You can import an existing global table in your stack by using `from` functions. You would need to specify
+ either table name, table arn or table attributes to import it.
+
+```typescript
+GlobalTable.fromTableName(stack, 'FooTableId', 'FooTable');
+```
+
+### NOTE
+
+* Be careful while naming your global table. If there is a name conflict in a region with a DynamoDB table,
+that could lead to removal of that table with the global table replica.
```

Comment:
WAIT WTF AGAIN

This needs to be up top when you talk about naming.

Also the scenario isn't quite clear to me.

Reply:
Looks like this was just for resources managed by CFN template. :)